### ISSUE TITLE
[shellscript] use Unix eol by default 

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#49418]
- VSCode Version:
- OS Version:

Steps to Reproduce:

1.
2.

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by Kroc** [comment]
In general; the ability to set line-ending settings per-language / file-type in settings, please.

**Comment by Tyriar** [comment]
@Kroc already exists, these are the default yaml settings:

```json
{
  "[yaml]": {
    "editor.insertSpaces": true,
    "editor.tabSize": 2,
    "editor.autoIndent": false
  }
}
```

**Comment by Kroc** [comment]
@Tyriar That's not line-endings though (LF / CRLF) which are **not** available per language!

**Comment by Tyriar** [comment]
@Kroc ah yeah you're right, the setting should be supported by language 👍

**Comment by aeschli** [comment]
Blocked by #39051

**Comment by DarthSpock** [comment]
Possibly not related but is there anything from the Notepad team you can borrow from for the EOL/CRLF since they added support for Linux, Mac, etc.?

**Comment by Tyriar** [comment]
@DarthSpock we've supported it since we launched, this is just about changing the default EOL when editing .sh files.

**Comment by DarthSpock** [comment]
Ok, I didn't catch that from anything I read here. Naturally it's already supported since you can run VS on any platform. Since Notepad is specific to Windows, I thought they might have something that make it more "Windows friendly". That does sound different from changing the default. I'm assuming the default would then change only if running in a Linux shell such as bash. A user should be able to change it of course but automating it would probably be of more use as it's one less thing to worry about.

**Comment by r1m** [comment]
A nice to have would be to follow by default `.gitattributes` values.

**Comment by aeschli** [comment]
Verify:
- set `files.eol` to `\r\n`
- open empty document, save as foo.sh: Verify that eol is set to '\n' (Check on the status bar)
- open empty document, switch to SH on the status bar. Verify that eol is set to '\n' (Check on the status bar)
